### PR TITLE
Replace bitwise flipping with WI_ flag helpers

### LINKED ISSUE #4026: Replace terminal parser/adapter bitwise flipping with WI_ flag helpers
I theoretically could replace all the flag flipping down this file with `WI_SetFlag` and friends. Thoughts?

_Originally posted by @miniksa in #3956

### PR DESCRIPTION
**Pull request by skyline75489**
## PR Checklist
* [X] Closes #4026

I didn't found much to do in `InputStateMachineEngine.cpp`, though.

### CODE REVIEW

File: src/terminal/adapter/adaptDispatchGraphics.cpp [hunk 1]
```diff
@@ -50,17 +50,12 @@ void AdaptDispatch::s_ApplyColors(WORD& attr, const WORD applyThis, const bool i
     // Mask off only the foreground or background
     if (isForeground)
     {
-        attr &= ~(FOREGROUND_BLUE | FOREGROUND_GREEN | FOREGROUND_RED | FOREGROUND_INTENSITY);
-        wNewColors &= (FOREGROUND_BLUE | FOREGROUND_GREEN | FOREGROUND_RED | FOREGROUND_INTENSITY);
+        WI_UpdateFlagsInMask(attr, FOREGROUND_BLUE | FOREGROUND_GREEN | FOREGROUND_RED | FOREGROUND_INTENSITY, wNewColors);
```
Comment (skyline75489) [inline src/terminal/adapter/adaptDispatchGraphics.cpp]: I check the source code of `WI_UpdateFlagsInMask` and I believe this is the right way to use it.

File: src/terminal/adapter/adaptDispatchGraphics.cpp [hunk 2]
```diff
@@ -24,11 +24,11 @@ void AdaptDispatch::s_DisableAllColors(WORD& attr, const bool isForeground)
 {
     if (isForeground)
     {
-        attr &= ~(FOREGROUND_BLUE | FOREGROUND_GREEN | FOREGROUND_RED | FOREGROUND_INTENSITY);
+        WI_ClearFlag(attr, FOREGROUND_BLUE | FOREGROUND_GREEN | FOREGROUND_RED | FOREGROUND_INTENSITY);
```
Comment (miniksa) [inline src/terminal/adapter/adaptDispatchGraphics.cpp]: I believe these have to be `WI_ClearAllFlags`.

File: src/terminal/adapter/adaptDispatchGraphics.cpp (hunk 1 above)
Comment (miniksa) [reply src/terminal/adapter/adaptDispatchGraphics.cpp]: Yeah that looks correct to me.

File: src/terminal/adapter/adaptDispatchGraphics.cpp [hunk 3]
```diff
@@ -106,37 +101,37 @@ void AdaptDispatch::_SetGraphicsOptionHelper(const DispatchTypes::GraphicsOption
...
     case DispatchTypes::GraphicsOptions::ForegroundGreen:
         s_DisableAllColors(attr, true); // turn off all color flags first.
-        attr |= FOREGROUND_GREEN;
+        WI_SetFlag(attr, FOREGROUND_GREEN);
         _changedForeground = true;
         break;
     case DispatchTypes::GraphicsOptions::ForegroundCyan:
         s_DisableAllColors(attr, true); // turn off all color flags first.
-        attr |= FOREGROUND_BLUE | FOREGROUND_GREEN;
+        WI_SetFlag(attr, FOREGROUND_BLUE | FOREGROUND_GREEN);
```
Comment (miniksa) [inline src/terminal/adapter/adaptDispatchGraphics.cpp]: Probably needs SetAllFlags (and below anywhere there's >1)

Comment (skyline75489) [reply src/terminal/adapter/adaptDispatchGraphics.cpp]: It seems to me that SetAllFlags should be used when flag is a variable. And SetFlag should be used when flag is compile time constant.

Comment (miniksa) [reply src/terminal/adapter/adaptDispatchGraphics.cpp]: I hear what you're saying, but WIL literally defines `WI_SetFlag` as `WI_SetAllFlags` with an assert about only setting one bit at a time. This is probably the reason why your builds are failing.

Comment (skyline75489) [reply src/terminal/adapter/adaptDispatchGraphics.cpp]: I misunderstood it. Force pushed the fix.

**Comment by skyline75489** [comment]
The CI failure is not related to the PR. Looks like there's timeout and it got cancelled.

**Comment by miniksa** [comment]
> The CI failure is not related to the PR. Looks like there's timeout and it got cancelled.

I went in and hit "re-run failed jobs". I don't know why it's being flaky. It looks like it got to completion then just sat there.

File: src/terminal/adapter/adaptDispatchGraphics.cpp (hunk 1 above)
Comment (DHowett-MSFT) [inline src/terminal/adapter/adaptDispatchGraphics.cpp]: oh, i love this

Comment (DHowett-MSFT) [reply src/terminal/adapter/adaptDispatchGraphics.cpp]: i feel like we may actually have a constant laying around for that mask. If not, should we?

Comment (DHowett-MSFT) [reply src/terminal/adapter/adaptDispatchGraphics.cpp]: it turns out we do: `FG_ATTRS` and `BG_ATTRS` and `META_ATTRS`

It's a define and not a constant, though, but that should be fine.

**Comment by DHowett-MSFT** [comment]
@msftbot merge this in 2 days

File: src/terminal/adapter/adaptDispatchGraphics.cpp (hunk 1 above)
Comment (skyline75489) [reply src/terminal/adapter/adaptDispatchGraphics.cpp]: Fixed. Now it's finally easy on the eyes.

**Comment by DHowett-MSFT** [comment]
@msftbot merge this in 2 minutes

**Comment by DHowett-MSFT** [comment]
:tada: Once again, thanks for the contribution!

This pull request was included in a set of conhost changes that was just
released with [Windows Insider Build 19603](https://blogs.windows.com/windowsexperience/2020/04/08/announcing-windows-10-insider-preview-build-19603/).

**Pull request merged**